src/scheduler/server_info.c

```diff
@@ -361,7 +362,7 @@ query_server(status *pol, int pbs_sd)
 		allcts = find_alloc_counts(sinfo->alljobcounts, PBS_ALL_ENTITY);
 		if (sinfo->alljobcounts == NULL)
 			sinfo->alljobcounts = allcts;
-
+		job_arrays_associated = TRUE;
```

Comment:
In the loop right below this function I call associate_parent_job function. However, control goes to this block only when there are limits set in the system. To avoid looping on running jobs twice when limit is set, I set this flag here and check for it outside. If no limits were set then I loop on running jobs to associate parent jobs.